src/main/java/wsu/edu/SnakePane.java

```diff
@@ -1,117 +1,239 @@
     }
     return snake;
   }
-  private void drawSnake(List<Circle> snake, Pane pane ){
+  private void drawSnake(List<Circle> snake) throws IOException {
     pane.getChildren().addAll(snake);
+    addFruit();
   }
-  private void moveSnakeRight(List<Circle> snake){
+
+  private void addFruit() throws IOException {
+      if (numFruits >= 0) {
+        System.out.print(pane.getMaxWidth() + ", " + pane.getMaxHeight());
+        fruit = new Circle((Math.random() * 600) + 10, (Math.random() * 400) + 10, 10, fruitColor);
+        pane.getChildren().add(fruit);
+        numFruits--;
+      }
+      else {
+        GameOver a = new GameOver();
+        a.gameOver(mainStage);
+      }
+
+  }
+  private void moveSnakeRight(List<Circle> snake, Timeline s) throws IOException {
     Circle oldHead = snake.get(0);
     Circle newHead = snake.remove(snake.size() - 1);
-    oldHead.setFill(Color.LIMEGREEN);
-    newHead.setFill(Color.RED);
-    newHead.setCenterX((oldHead.getCenterX() + 20) % 600);
+    oldHead.setFill(snakeBodyColor);
+    newHead.setFill(snakeHeadColor);
+    newHead.setCenterX((oldHead.getCenterX() + 21) % 600);
     newHead.setCenterY(oldHead.getCenterY());
     snake.add(0, newHead);
+    if(checkColision(snake)) s.stop();
   }
-  private void moveSnakeLeft(List<Circle> snake){
+  private void moveSnakeLeft(List<Circle> snake, Timeline s) throws IOException {
     Circle oldHead = snake.get(0);
     Circle newHead = snake.remove(snake.size() - 1);
-    oldHead.setFill(Color.LIMEGREEN);
-    newHead.setFill(Color.RED);
-    newHead.setCenterX((oldHead.getCenterX() - 20 + 600) % 600);
+    oldHead.setFill(snakeBodyColor);
+    newHead.setFill(snakeHeadColor);
+    newHead.setCenterX((oldHead.getCenterX() - 20 + 599) % 600);
     newHead.setCenterY(oldHead.getCenterY());
     snake.add(0, newHead);
+    if(checkColision(snake)) s.stop();
   }
-  private void moveSnakeDown(List<Circle> snake){
+  private void moveSnakeDown(List<Circle> snake, Timeline s) throws IOException {
     Circle oldHead = snake.get(0);
     Circle newHead = snake.remove(snake.size() - 1);
-    oldHead.setFill(Color.LIMEGREEN);
-    newHead.setFill(Color.RED);
-    newHead.setCenterY((oldHead.getCenterY() + 20) % 400);
+    oldHead.setFill(snakeBodyColor);
+    newHead.setFill(snakeHeadColor);
+    newHead.setCenterY((oldHead.getCenterY() + 21) % 400);
     newHead.setCenterX(oldHead.getCenterX());
     snake.add(0, newHead);
+    if(checkColision(snake)) s.stop();
   }
-  private void moveSnakeUp(List<Circle> snake){
+  private void moveSnakeUp(List<Circle> snake, Timeline s) throws IOException {
     Circle oldHead = snake.get(0);
     Circle newHead = snake.remove(snake.size() - 1);
-    oldHead.setFill(Color.LIMEGREEN);
-    newHead.setFill(Color.RED);
-    newHead.setCenterY((oldHead.getCenterY() - 20 + 400) % 400);
+    oldHead.setFill(snakeBodyColor);
+    newHead.setFill(snakeHeadColor);
+    newHead.setCenterY((oldHead.getCenterY() - 20 + 399) % 400);
```

Comment:
The modulo should be removed from the game -- this is going to make it hard to detect when the snake is off the pane.